credentials/src/main/kotlin/SSI.kt

```diff
@@ -270,61 +270,115 @@ public class VerifiablePresentation private constructor() {
+ * descriptor is not satisfied by the provided credentials, an exception is thrown.
  *
- * @return the selectable credentials.
+ * ### Throws
+ * - [NotImplementedError] if the Presentation Definition contains Submission Requirements or if a Field Filter is included.
+ * - [Exception] if any required field is not satisfied in the given InputDescriptor.
+ *
+ * @param presentationDefinition The [PresentationDefinitionV2] object representing the presentation's definition.
+ * @param vcJwts The list of [VcJwt] representing the Verifiable Credentials to be validated against the presentation definition.
+ * @throws Exception If a required field is not satisfied in a given InputDescriptor.
  */
 @Throws(Exception::class)
-private fun selectFrom(presentationDefinition: PresentationDefinitionV2, vcJwts: List<VcJwt>): List<VcJwt> {
-  val selectableCredentials = mutableListOf<VcJwt>()
-
+private fun validatePresentationFrom(presentationDefinition: PresentationDefinitionV2, vcJwts: List<VcJwt>) {
   if (!presentationDefinition.submissionRequirements.isNullOrEmpty()) {
     throw NotImplementedError("Presentation Definition's Submission Requirements feature is not implemented")
-  } else {
-    for (inputDescriptor: InputDescriptorV2 in presentationDefinition.inputDescriptors) {
-      // Fields Processing
-      if (inputDescriptor.constraints.fields!!.isNotEmpty()) {
-        for (vcJwt: VcJwt in vcJwts) {
-          var fieldMatch = false
-
-          for (field: FieldV2 in inputDescriptor.constraints.fields) {
-
-            // Optional fields are not needed to complete the required fields in the presentation definition
-            if (field.optional != null && field.optional) {
-              continue;
-            }
+  }
 
-            for (path: String in field.path) {
-              val jsonPathResult = evaluateJsonPath(vcJwt, path)
+  for (inputDescriptor: InputDescriptorV2 in presentationDefinition.inputDescriptors) {
+    if (inputDescriptor.constraints.fields!!.isNotEmpty()) {
+      for (field: FieldV2 in inputDescriptor.constraints.fields) {
+        if (field.optional == true) continue // Skip optional fields
 
-              if (jsonPathResult != null) {
-                if (field.filter != null) {
-                  throw NotImplementedError("Field Filter is not implemented")
-                } else {
-                  fieldMatch = true
-                  break
-                }
+        var fieldMatch = false
+        for (vcJwt: VcJwt in vcJwts) {
+          for (path: String in field.path) {
+            val jsonPathResult = evaluateJsonPath(vcJwt, path)
+            if (jsonPathResult != null) {
+              if (field.filter != null) {
+                throw NotImplementedError("Field Filter is not implemented")
+              } else {
+                fieldMatch = true
+                break
               }
             }
           }
```

Comment:
Seems like this could be simplified with [map](https://kotlinlang.org/docs/collection-transformations.html#map) and [filter](https://kotlinlang.org/docs/collection-filtering.html#filter-by-predicate) (or maybe [first](https://kotlinlang.org/docs/collection-elements.html#retrieve-by-condition))?

Reply:
I think leaving it explicit is ok because there is a lot more that will be added as new features are added.

All of these not implemented forks will be implemented shortly